### ISSUE TITLE
Extreme Benchmarking - Modifying/adding ClaymoreDual miner files for higher profit

### ISSUE THREAD

**Issue by fonyo** [aaronsace/MultiPoolMiner#111]
# SCROLL DOWN TO THE BOTTOM OF THIS PAGE TO SEE HOW TO MODIFY THE MINER FILES

![img_0546](https://cloud.githubusercontent.com/assets/10072275/26283058/edeb504c-3e17-11e7-8069-120c77e729ac.JPG)
2x RX 480 nitro+ 8GB

**Comment by fonyo** [comment]
current revenue: 11.7 usd per day

**Comment by fonyo** [comment]
My problem is, even though dcri 29 is the most profitable, MPM only runs 28. Why is that so?

**Comment by fonyo** [comment]
Also, what is the switch threshold and where can I adjust it please?

**Comment by aaronsace** [comment]
The difference is infinitesimal; there is a delta applied to stop needless switching.

**Comment by fonyo** [comment]
Can you make it editable through the start.bat please? I do agree with the needless switching hence my other issue regarding problems with weaker cards and switching. I would like to be able to set the fraction of percentage myself regardless just like you can do with nicehash.

**Comment by fonyo** [comment]
Basically if you solve these two issues, you will crack and make a better app than nicehash did.

**Comment by imrecsoka** [comment]
Why do i feel so behind here, i dont even really understand that benchmarking? Please somebody guide me here i really want to also do the right thing. I have 2 graphics cards also according to Lee's calculations it says i only get 6 USD a day, what is wrong with my life here?
Appreciate help here please? ill mine for you for 2 bloody hours if i have to... please guide me

**Comment by fonyo** [comment]
I started copy-pasting the claymoredual30 file from the miners folder and every time I did I changed the number 30 (which occurs 3 times in the file including the dcri value itself plus the file name) to another dcri value and saved it in the miners folder. So now I have got not only claymore solo, dual20, dual30 and dual40 but one for each dcri value between 18 and 40 to get the most accurate and profitable mining speed. It was very tedious and took ages to get the bench done but now MPM will relaunch the miner according to the best revenue (correct dcri where ethash rev + sia rev = largest revenue). It obviously requires a lot of patience but it was totally worth it. Please note the only reason I did it because Sia had a huge price hike and there were times when it was more profitable than any other ethash algo. Most of the time Sia only makes spare change and a lot of electricity bill and heat.

**Comment by imrecsoka** [comment]
Please send me the file i want to see what u mean, i dont get it.

**Comment by imrecsoka** [comment]
I saw a spike in SIA ETHER ZCASH SKEIN and MYRIAD but i dont understand how to make the program mine them and if they work they have failures.
Its like some of those dont like AMD which i find hard to believe.... mining is more popular now so surely more dev has been done.

**Comment by imrecsoka** [comment]
I have installed 2013 vc++ x32 and x64 but i still get dll errors.

**Comment by imrecsoka** [comment]
** cpuminer-multi 1.3 by tpruvot@github **
BTC donation address: 1FhDPLPpw18X4srecguG3MxJYe4a1JsZnd (tpruvot)

[2017-05-23 01:26:44] 8 miner threads started, using 'sia' algorithm.
[2017-05-23 01:26:44] Starting Stratum on stratum+tcp://hub.miningpoolhub.com:17
022
[2017-05-23 01:26:45] Stratum answer id is not correct!
[2017-05-23 01:28:13] stratum_recv_line failed
[2017-05-23 01:28:13] Stratum connection interrupted
[2017-05-23 01:28:14] Stratum answer id is not correct!
[2017-05-23 01:29:08] stratum_recv_line failed
[2017-05-23 01:29:08] Stratum connection interrupted
[2017-05-23 01:29:09] Stratum answer id is not correct!
[2017-05-23 01:30:03] stratum_recv_line failed
[2017-05-23 01:30:03] Stratum connection interrupted
[2017-05-23 01:30:03] Stratum answer id is not correct!
[2017-05-23 01:30:57] stratum_recv_line failed
[2017-05-23 01:30:57] Stratum connection interrupted
[2017-05-23 01:30:58] Stratum answer id is not correct!
[2017-05-23 01:31:55] stratum_recv_line failed
[2017-05-23 01:31:55] Stratum connection interrupted

**Comment by fonyo** [comment]
I would leave myriad-groestl and skein alone. They are up and down, usually by the time they are cashed in they worth way less so you lose a lot. Configs/settings are also very scarce on the internet so I don't even know where to start with them. I would also leave CPU mining alone as it's not really worth it unless you have a shiny new Ryzen CPU. I'm not at my rig so I can't show you the modifications just yet re claymore-ethash miner config files.

**Comment by fonyo** [comment]
@imrecsoka 
So the way I did it. Here's a link to the dual30 miner config file. You have these in the Miners folder on your computer. 
https://github.com/aaronsace/MultiPoolMiner/blob/master/Miners/ClaymoreDual30.txt

Line 4 and 5, I highlighted below (bold text) what needs to be changed to 17 and save it as ClaymoreDual17.txt
"Arguments":  "-epool $($Pools.Ethash.Host):$($Pools.Ethash.Port) -ewal $($Pools.Ethash.User) -epsw $($Pools.Ethash.Pass) -esm 3 -allpools 1 -dpool $($Pools.Sia.Host):$($Pools.Sia.Port) -dwal $($Pools.Sia.User) -dpsw $($Pools.Sia.Pass) -dcoin sc -dcri **30**",
    "HashRates":  {"Ethash":  "$($Stats.ClaymoreDual**30**_Ethash_HashRate.Week)","Sia": "$($Stats.ClaymoreDual**30**_Sia_HashRate.Week)"},

Then do the same thing for 18, 19, 21, 22, 23, 24, 25, 26, 27, 28, 29, 31, 32, 33, 34, 35, 36, 37, 38, 39.
There is one for solo, dual20, dual30 and dual40 already.

To sum it up, each time you have to change those 3 numbers plus the filename to another dcri value.
Make sure they all match, otherwise you are going to get weird stats. During benchmarking if MPM doesn't switch after 1 minute, that indicates you made an error in that miner file and benchmarking cannot carry on (happened to me twice). It is very easy to miss one of them. 

TL;DR

Basically, the dcri parameter determines how aggressively you are pushing the secondary coin onto your GPU. The more you push it, the less Ethash you will get. If Sia continues to drop in price, it might worth adding 16, 15, 14 etc. We'll see because lowering the dcri value below a certain number will not increase the ethash but only decreases the hash for the secondary coin (Sia in this case).

You can skip creating the ones above 30. I don't think Sia's price will go that high any time soon.

This may or may not apply to other secondary coins the exact same way such as Decred, LBRY or Pascal. I have never tested them because MPH doesn't have these coins. If MPH adds support for Decred, the same amount of files need to be created for Decred as well in order to get an accurate picture and most profitable hash to hash ratio.

I hope this makes sense.
